### ISSUE TITLE
Cannot set lag of NamedControl in NodeProxy externally

### ISSUE THREAD

**Issue by madskjeldgaard** [supercollider/supercollider#5709]
## Environment

* SuperCollider version: 3.12.2
* Operating system: Arch
* Other details (Qt version, audio driver, etc.): Running in SCNVIM

## Steps to reproduce

(Example from PitchTrebler at https://scsynth.org/t/live-streaming-with-mads/5259/2)

This works:

```supercollider
Ndef.clear;
Ndef(\a, {|freq = 200, amp = 0.2| Saw.ar(freq, amp!2)}).lag(\freq, 5, \amp, 2).play;
Ndef(\a).set(\freq, 100, \amp, 0.05);
```
With namedcontrols it does not:

```supercollider
Ndef.clear;
Ndef(\a, {Saw.ar(\freq.kr(200), \amp.kr(0.2)!2)}).lag(\freq, 5, \amp, 2).play;
Ndef(\a).set(\freq, 100, \amp, 0.05);
```

## Expected vs. actual behavior

I would expect to be able to set the lag on a NamedControl in the same way as other parameters in a NodeProxy. 

PS. I could not find an issue on this topic even though it seems like a widely known thing? See for example https://scsynth.org/t/live-streaming-with-mads/5259/3?u=madskjeldgaard

**Comment by madskjeldgaard** [comment]
Not sure if this is perhaps related https://github.com/supercollider/supercollider/pull/2127

**Comment by telephon** [comment]
Yes, it is one of those things that are almost deprecated (see below). I had always tried to expose all things in the underlying objects, like SynthDef here. But because NamedControls is a bit orthogonal to the SynthDef rates, this doesn't work. There is no dedicated control name for it either, so to would at least be a bit of trickery.

```supercollider
setRates { |args|
		//this.deprecated(thisMethod); // maybe deprecate later
		rates = rates ?? { IdentityDictionary.new };
		rates.putPairs(args);
	}
```

**Comment by madskjeldgaard** [comment]
It also affects `SynthDef.wrap` it seems:

```supercollider
f = {|freq=110| SinOsc.ar(freq)}
Ndef(\nolag, { SynthDef.wrap(f) }).play;
Ndef(\nolag).lag(\freq, 10);
Ndef(\nolag).set(\freq, 220)
```

**Comment by madskjeldgaard** [comment]
> Yes, it is one of those things that are almost deprecated (see below). I had always tried to expose all things in the underlying objects, like SynthDef here. But because NamedControls is a bit orthogonal to the SynthDef rates, this doesn't work in the second case. There is no dedicated control name for it either, so to would at least be a bit of trickery.
> 
> ```supercollider
> setRates { |args|
> 		//this.deprecated(thisMethod); // maybe deprecate later
> 		rates = rates ?? { IdentityDictionary.new };
> 		rates.putPairs(args);
> 	}
> ```

Thanks for the response Julian! Not sure I understand - is NamedControl deprecated?

**Comment by telephon** [comment]
No, not at all! NamedControl came later and only partially replaced some features that are built into SynthDef creation. 

The `setRates` code comes from NodeProxy, my intuiton was at some stage that nobody uses that. Well, you never know ;)

**Comment by jamshark70** [comment]
I think this stems from the fact that SynthDef initially had no standardized way to query or manipulate the details of control names. Now we are finding use cases where it's helpful or necessary to standardize and unify the representation.

So I think this should probably be discussed in an RFC: Design a data structure and interface that meets the requirements first.

My opinion remains that SynthDef is the right place for this. But there was some work on another issue where control names were being collected under the banner of NamedControl. I still disagree with that but perhaps the ship already sailed.

**Comment by telephon** [comment]
> I still disagree with that but perhaps the ship already sailed.

I agree that all that should be in `SynthDef`. In terms of implementation, it could be moved there easily (I suggested this recently). But yes, the whole thing needs a unified treatment, that would be good.
